distribution/lib/Standard/Base/0.0.0-dev/src/Data/Index_Sub_Range.enso

```diff
@@ -149,8 +149,9 @@ sort_and_merge_ranges ranges =
      the method wants to work only with normalized ranges, it must perform
      normalization on its own.
    - range: The `Index_Sub_Range` to take from the collection.
-take_helper : Integer -> (Integer -> Any) -> (Integer -> Integer -> Any) -> (Vector (Integer | Range) -> Vector Any) -> Index_Sub_Range -> Any
+take_helper : Integer -> (Integer -> Any) -> (Integer -> Integer -> Any) -> (Vector (Integer | Range) -> Vector Any) -> (Index_Sub_Range | Range | Integer) -> Any
 take_helper length at single_slice slice_ranges index_sub_range = case index_sub_range of
+    _ : Integer -> single_slice 0 (Math.min length index_sub_range)
```

Comment:
```suggestion
    count : Integer -> single_slice 0 (Math.min length count)
```

To me it is confusing when `index_sub_range` is used as an integer - IMO the rename would make things clearer here.

Also wouldn't it be easier to delegate to the existing implementation instead to DRY?

```suggestion
    count : Integer -> take_helper length at single_slice slice_ranges (First count)
```